hack/local-setup.sh

```diff
@@ -46,6 +47,34 @@ deployIngressController () {
   kubectl -n ingress-nginx wait --timeout=300s --for=condition=Available deployments --all
 }
 
+deployMetalLB () {
+  clusterName=${1}
+  metalLBSubnet=${2}
+
+  kubectl config use-context kind-${clusterName}
+  echo "Deploying MetalLB to ${clusterName}"
+  ${KUSTOMIZE_BIN} build ${METALLB_KUSTOMIZATION_DIR} | kubectl apply -f -
+  echo "Waiting for deployments to be ready ..."
+  kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=90s
+  echo "Creating MetalLB AddressPool"
+  cat <<EOF | kubectl apply -f -
+apiVersion: metallb.io/v1beta1
+kind: IPAddressPool
+metadata:
+  name: example
+  namespace: metallb-system
+spec:
+  addresses:
+  - 172.32.${metalLBSubnet}.0/24
+---
+apiVersion: metallb.io/v1beta1
+kind: L2Advertisement
+metadata:
+  name: empty
+  namespace: metallb-system
```

Comment:
just curious, what is this for?

Reply:
Good question.
I tried these changes locally without the L2Advertisement CR.
The curl request to the service loadbalancer IP Address times out.

My understanding is this resource instructs metallb to use Layer 2 protocols (ARP in this case) to advertise that the IP address is available on the network. ARP (Address Resolution Protocol) is used by L2 devices/interfaces to discover the MAC address for a given IP so it can construct the appropriate L2/Ethernet frame header with the destination MAC address.

From https://metallb.universe.tf/concepts/
> After MetalLB has assigned an external IP address to a service, it needs to make the network beyond the cluster aware that the IP “lives” in the cluster. MetalLB uses standard networking or routing protocols to achieve this, depending on which mode is used: ARP, NDP, or BGP.

BGP is also supported, but I haven't tried that (I'm also not sure if it's necessary as I believe it's kind of a flat network that's established, so no routing is required)

Reply:
Thanks for looking that up! It's actually very little setup to have l4 load balancers working in local dev envs, very interesting!

Reply:
Not a big deal, but could these resources be added to the kustomize config itself instead of applying them directly here. The use of the `metalLBSubnet` probably makes that harder, but it would tidy this up a bit.

Reply:
Capturing the outcome of offline conversation.
We've used the `replacements` field before in kustomize, but that pattern doesn't apply here.
Generally, kustomize doesn't help when working with variables (deliberately so, as it pushes for file based configuration).
It would be possible to have a base IPAddressPool resource in the config dir, and then use yq to overwrite `spec.addresses` for each install of metallb. However, given this change is almost certainly going to be local dev only (load balancers are available in cloud providers), that change doesn't offer anything additional outside of what it already does.